src/hooks/useFavorites.ts

```diff
@@ -21,5 +28,5 @@ export function useFavorites() {
     setFavorites(updated);
   };
 
-  return [favorites, toggleFavorite];
+  return [favorites, toggleFavorite] as const;
```

Comment:
These `as const`s are key for telling TypeScript that this isn't any old mutable array, this is an immutable fixed-size tuple.

```diff
- (string | ((name: string) => void)[]
} [string, (name: string) => void]
```